Mathlib/Data/Fin/Tuple/Sort.lean

```diff
@@ -107,6 +109,46 @@ theorem monotone_sort (f : Fin n → α) : Monotone (f ∘ sort f) := by
   exact (monotone_proj f).comp (graphEquiv₂ f).monotone
 #align tuple.monotone_sort Tuple.monotone_sort
 
+lemma Fintype.card_fin_lt_nat (m g : ℕ) (h : g ≤ m) : Fintype.card {i : Fin m // i < g } = g := by
+  conv_rhs => rw [← Fintype.card_fin g]
+  apply Fintype.card_congr
+  exact ⟨ fun x => ⟨x, x.prop⟩, fun x => ⟨⟨x, (lt_of_lt_of_le (Fin.is_lt x) h)⟩, x.prop⟩,
+    fun x => by simp, fun x => by simp⟩
+
+/-- A sorted tuple with `m` elements and exactly `Fintype.card {i // f i ≤ a}` less than `a`, has
+the elements at the start, and vice versa -/
+theorem lt_card_le_iff_apply_le_of_monotone {α} [LinearOrder α] (m : ℕ) (f : Fin m → α) (a : α)
```

Comment:
Is `LinearOrder` still needed, or is this now true on partial orders?

Reply:
I don't think partial order will work here. All the elements must be comparable with each other. Partial order allows non-comparability for some cases

Reply:
It works fine, with minimal proof changes